GVFS/GVFS.UnitTests/Common/GVFSEnlistmentTests.cs

```diff
@@ -0,0 +1,42 @@
+﻿using GVFS.Common.Git;
+using GVFS.Tests.Should;
+using GVFS.UnitTests.Mock.Common;
+using GVFS.UnitTests.Mock.Git;
+using NUnit.Framework;
+
+namespace GVFS.UnitTests.Common
+{
+    [TestFixture]
+    public class GVFSEnlistmentTests
+    {
+        public const string MountId = "85576f54f9ab4388bcdc19b4f6c17696";
+        public const string EnlistmentId = "520dcf634ce34065a06abaa4010a256f";
+
+        [TestCase]
+        public void CanGetMountId()
+        {
+            MockGVFSEnlistment enlistment = this.CreateEnlistment();

```

Comment:
It's a little bit odd to use a mock object as the object under test. The more pure answer would be to update `GVFSEnlistment` to take a factory method as an argument, but that is overkill here, and would unnecessarily complicate the product code.

You might consider creating a private class here named `TestGVFSEnlistment` that derives from `GVFSEnlistment` and use that instead, so that future enhancements to the common mock object don't interfere with these unit tests.

Reply:
@sanoursa you're correct here.  I created my own TestGVFSEnlistment, so we're protected against future changes to the MockGVFSEnlistment.